README.md

```diff
@@ -1,15 +1,68 @@
+
+This implementation of the ZMLT algorithm uses three layout algorithms - [BatchTree](https://github.com/cns-iu/map4sci-BatchTree), [CG](https://github.com/cns-iu/map4sci-CG), and [DELG](https://github.com/cns-iu/map4sci-DELG) - to generate layouts for the input graphs. These algorithms are known for their efficiency and ability to handle large graphs.
+
+The ZMLT algorithm also provides semantic zoom functionality, which enables users to zoom in and out of the graph while maintaining the integrity of the graph structure. This functionality is especially useful when exploring large graphs that have multiple levels of detail.
+
+This implementation of the ZMLT algorithm is designed to be user-friendly and easy to use. We hope that this project will be a useful resource for researchers, developers, and anyone who needs to visualize large graphs in an interactive and efficient way.
 
-This algorithm uses the [BatchTree implementation](https://github.com/khaled-rahman/BatchTree) of the ZMLT layout algorithm.
 
 ## Quick Start
 
 Users of map4sci should use the map4sci-project-template as a template for their data projects. See <https://github.com/cns-iu/map4sci-project-template> for more information.
 
+# Creating Dataset
+
+To create the dataset, follow these steps:
+
+1. Create a folder with the name of the dataset inside map4sci/data-processor/dataset.
+2. Inside the folder, create a `config.sh` file and a `network.dot` file.
+
+## Configuring the Dataset
+
+The config.sh file contains the configuration parameters for the dataset, which are used by the layout algorithm to generate the visualization. Here's a brief overview of the parameters:
+
+All the `config.sh` files should follow the format as given in [config.example.sh](data-processor/datasets/config.example.sh)
+
+The `network.dot` file plays a crucial role in generating the graph layout, as it provides input graph data to the ZMLT algorithm. Nodes and edges in the network.dot file represent entities or objects in the graph, while labels associated with them provide additional information that can be used to interpret the graph. Levels defined in the network.dot file define the hierarchy of the graph, which enables the ZMLT algorithm to visualize the graph with multiple levels of detail.
+
+Creating and editing the network.dot file can be done manually using a text editor or generated automatically using a script or graph visualization tool. Its importance to the ZMLT algorithm implementation cannot be overstated, as it serves as the backbone for generating visually appealing and informative graph layouts.
+
+## Running using Docker
+
+1. Install Docker 
+2. Clone the Map4Sci repository from GitHub
+3. cd map4sci
+4. Build the Docker image: docker build -t map4sci .
+5. Start a Docker container from the image: docker run -it map4sci 
```

Comment:
You need to test this as it will run the run.sh script automatically when you do this. Add instructions on the options you can include here (ie CURRENT_DATASET, CURRENT_VERSION, NODE_OPTIONS, etc.)